src/graph.ts

```diff
@@ -164,4 +163,36 @@ export const sanitizeGraph = (graph: Graph) => {
       node.value = -1;
     }
   }
-}
\ No newline at end of file
+};
+
+export const willExitAnOwnMarble = (
```

Comment:
The function looks good to me, but the logic is very similar to moveMarbleInDirection.

Maybe we should merge the two, to avoid unnecessary loop.